main/src/ca/uwaterloo/flix/api/Bootstrap.scala

```diff
@@ -82,5 +117,35 @@ class Bootstrap {
 
   }
 
+  private def createPathList(path: Path, ext: String): List[Path] = {
+    val file = path.toFile
+    val files = file.listFiles()
+    if(files == null) {
+      List.empty
+    } else {
+      files.toList.filter(file =>
+        if (file.isFile) {
+          val i = file.toString.lastIndexOf('.')
+          val extension = if (i > 0) file.toString.substring(i + 1) else ""
```

Comment:
This is pretty low-level, check out `Packager.isJarFile` I think that is the preferred way.